subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/vcs/VcsDependencyResolver.java

```diff
@@ -55,6 +57,7 @@
     private final VcsMappingFactory vcsMappingFactory;
     private final VersionControlSystemFactory versionControlSystemFactory;
     private final File baseWorkingDir;
+    private final Map<String, VersionRef> latestIntegrationVersionCache = new HashMap<String, VersionRef>();
```

Comment:
Let's call it `selectedVersionCache`

This isn't only for storing the versions resolved from `latest.integration`